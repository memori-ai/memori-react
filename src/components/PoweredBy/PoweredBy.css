.memori--powered-by {
    position: absolute;
    z-index: 1000;
    bottom: 1rem;
    left: 0;
    display: inline-flex;
    flex-wrap: wrap;
    align-items: center;
    padding: 0.25rem 0.5rem;
    border-radius: 10px;
    background-color: var(--memori-inner-bg, transparent);
    box-shadow: var(--memori-button-box-shadow);
    color: var(--memori-text-color, rgba(0, 0, 0, 0.85)) !important;
    font-size: 0.7em;
}

@media (max-width: 768px) {
    .memori--powered-by {
        top: 3.5rem;
        bottom: auto;
    }
}

.memori--powered-by img {
    max-width: 2rem;
    max-height: 2rem;
    margin-right: 0.25rem;
}

.memori--powered-by p {
    margin-top: 0;
    margin-bottom: 0;
    color: var(--memori-text-color, rgba(0, 0, 0, 0.85)) !important;
}

.memori--powered-by a {
    color: var(--memori-text-color, rgba(0, 0, 0, 0.85)) !important;
    text-decoration: underline;
}

.memori--powered-by a:hover,
.memori--powered-by a:focus {
    color: var(--memori-text-color, rgba(0, 0, 0, 0.85)) !important;
    opacity: 0.8;
    text-decoration: underline;
}