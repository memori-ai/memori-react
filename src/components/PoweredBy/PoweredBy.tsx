import { Tenant } from '@memori.ai/memori-api-client/dist/types';

export interface Props {
  tenant?: Tenant;
  userLang?: string;
  integrationID?: string;
  memoriHash?: string;
}

const PoweredBy = ({
  tenant,
  userLang = 'en',
  integrationID,
  memoriHash,
}: Props) => {
  const tenantId =
    tenant?.theme === 'memorytwin'
      ? 'memorytwin'
      : tenant?.theme === 'twincreator'
      ? 'twincreator'
      : 'aisuru';

  return (
    <div className="memori--powered-by">
      <img src={`https://aisuru.com/images/${tenantId}/logo.png`} alt="" />
      <p>
        <span className="sr-only">Powered by</span>
        <a
          href={`https://memori.ai/${
            userLang?.toLowerCase() === 'it' ? 'it' : 'en'
          }${integrationID ? `?integrationID=${integrationID}` : ''}${
            memoriHash ? `${integrationID ? '&' : '?'}memori=${memoriHash}` : ''
          }`}
          target="_blank"
          rel="noopener noreferrer"
        >
          Memori.AI
        </a>
      </p>
    </div>
  );
};

export default PoweredBy;
