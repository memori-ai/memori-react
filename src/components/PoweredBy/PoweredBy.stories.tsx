import { Meta, Story } from '@storybook/react';
import PoweredBy, { Props } from './PoweredBy';

import './PoweredBy.css';

const meta: Meta = {
  title: 'Powered by',
  component: PoweredBy,
  argTypes: {},
  parameters: {
    controls: { expanded: true },
  },
};

export default meta;

const Template: Story<Props> = args => <PoweredBy {...args} />;

export const Default = Template.bind({});
Default.args = {
  userLang: 'en',
};

export const Italian = Template.bind({});
Italian.args = {
  userLang: 'it',
};

export const MemoryTwin = Template.bind({});
MemoryTwin.args = {
  tenant: {
    id: 'memorytwin',
    theme: 'memorytwin',
    config: {
      name: 'MemoryTwin',
      showNewUser: true,
      requirePosition: false,
      feedbackURL: '',
    },
  },
};

export const TwinCreator = Template.bind({});
TwinCreator.args = {
  tenant: {
    id: 'twincreator',
    theme: 'twincreator',
    config: {
      name: 'twincreator',
      showNewUser: true,
      requirePosition: false,
      feedbackURL: '',
    },
  },
};

export const AIsuru = Template.bind({});
AIsuru.args = {
  tenant: {
    id: 'AIsuru',
    theme: 'aisuru',
    config: {
      name: 'AIsuru',
      showNewUser: true,
      requirePosition: false,
      feedbackURL: '',
    },
  },
};

export const WithData = Template.bind({});
WithData.args = {
  integrationID: 'abc',
  memoriHash: 'tenant-owner-memori',
};
