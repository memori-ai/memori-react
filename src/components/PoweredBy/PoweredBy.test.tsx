import React from 'react';
import { render } from '@testing-library/react';
import PoweredBy from './PoweredBy';

it('renders PoweredBy unchanged', () => {
  const { container } = render(<PoweredBy />);
  expect(container).toMatchSnapshot();
});

it('renders PoweredBy with data unchanged', () => {
  const { container } = render(
    <PoweredBy integrationID="abc" memoriHash="tenant-owner-memori" />
  );
  expect(container).toMatchSnapshot();
});
