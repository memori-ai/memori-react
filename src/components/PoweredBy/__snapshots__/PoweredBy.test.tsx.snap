// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders PoweredBy unchanged 1`] = `
<div>
  <div
    class="memori--powered-by"
  >
    <img
      alt=""
      src="https://aisuru.com/images/aisuru/logo.png"
    />
    <p>
      <span
        class="sr-only"
      >
        Powered by
      </span>
      <a
        href="https://memori.ai/en"
        rel="noopener noreferrer"
        target="_blank"
      >
        Memori.AI
      </a>
    </p>
  </div>
</div>
`;

exports[`renders PoweredBy with data unchanged 1`] = `
<div>
  <div
    class="memori--powered-by"
  >
    <img
      alt=""
      src="https://aisuru.com/images/aisuru/logo.png"
    />
    <p>
      <span
        class="sr-only"
      >
        Powered by
      </span>
      <a
        href="https://memori.ai/en?integrationID=abc&memori=tenant-owner-memori"
        rel="noopener noreferrer"
        target="_blank"
      >
        Memori.AI
      </a>
    </p>
  </div>
</div>
`;
